### ISSUE TITLE
GUI not in sync with zwave status Qubino ZMNHDD1

### ISSUE THREAD

**Issue by bueno79** [domoticz/domoticz#1726]
Version: 3.8063
Platform: Raspberry Pi 3
Plugin/Hardware: z-wave
Description: GUI not in sync with zwave status

After a restart (or crash) domoticz GUI is not in sync with the status of a few zwave devices. Domoticz initialize and start querying the zwave devices. Then a light (dimmer) turns on in the GUI. But the OpenZWave Control Panel shows that the current values the light is off. After manually disable the dimmer, it's enabled again and after disable it again, it's stayed off and everything is back to normal.

It looks like it's only happening with my Qubino ZMNHDD1 Flush Dimmers.

Open Zwave Control Panel shows the dimmer is Off:
![image](https://user-images.githubusercontent.com/27161702/28717938-147bd09e-73a4-11e7-8556-7bcdf9553f2c.png)

The domoticz GUI shows the dimmer is On:
![image](https://user-images.githubusercontent.com/27161702/28717951-1ff9bd5a-73a4-11e7-831f-a4487835f955.png)

Zwave associations:
![image](https://user-images.githubusercontent.com/27161702/28717957-2b83208a-73a4-11e7-9020-6dee9374bc78.png)

Also after a clean install still got the problems. Then I tried it with a stable version. Stable version 3.5877 has the same problems

regards

**Comment by gizmocuz** [comment]
Could you make a patch for this ?

**Comment by bueno79** [comment]
hi gizmocuz,

do you asking me if I could do a patch ? Yes maybe. I'm developer and in IT since 20 years now. If you explain a little bit how to get infos about changes I need to do.

**Comment by gizmocuz** [comment]
Yes that would be great.
First thing you might want to do is get your development environment setup,  i suggest to use microsoft visual studio
Then you will see in the solution explorer under 'devices', zwave, in this class add some breakpoints in the update-value function

**Comment by bueno79** [comment]
oh I see. You need toi debug the function update-value to understand why return is not acknowledge by domoticz. I don't have Visual Studio License, but more than this...I don't have skills to debug functions write in C.
Even, my domoticz is on a raspberry. VS can debug it remotely ?

**Comment by gizmocuz** [comment]
visual studio community is free...

**Comment by gizmocuz** [comment]
yes vs can debug remotely, but best to plug your zwave stick in a fast system and debug directly with visual studio (of course!)

**Comment by bueno79** [comment]
ok going to try cause I think you don't have a Qubino ZMNHDD1 Flush Dimmers. So If I'm not able to do it...it will be difficult for you to fix this right ? If I fail...what is the plan B ?

**Comment by gizmocuz** [comment]
correct.. one can not have all hardware that exists in the world today

**Comment by bueno79** [comment]
ok. I install VS community. Component apps in linux C++ ? right ?

**Comment by gizmocuz** [comment]
No.. just use the visual studio sollution (see the readme files in the repo folder) and windows...
If you have not windows... format your pc, use virtualbox or something else..

**Comment by bueno79** [comment]
just found a function COpenZwave::UpdateValue in OpenZwave.cpp. Are we talking about this function ?

**Comment by gizmocuz** [comment]
Yes.. you need to set some breakpoints and do some debugging there

**Comment by bueno79** [comment]
ok impossible to buld the project. The problem is Boost and VS 2017...fix so many error during the built of the library...but I cannot continue.
Maybe you should write an article : how to setup environnement to work on Domoticz. Really sorry. So maybe this problem will never be fix. Very sad this module is really small, fast, no bypass and no noise like many other dimmer.

if you have a solution and if I can help you, do not hesitate to contact me.

regards and thx for all your effort for Domoticz
Bueno

**Comment by gizmocuz** [comment]
Did you downloaded the windows libraries from our git repo ? (as stated in the install.txt file)

**Comment by gizmocuz** [comment]
Do you live in the Netherlands ? Maybe you can borrow me one ?

**Comment by ghost** [comment]
I can borrow you a unit. I think I have a spare one and live in the Netherlands. Pm me your address.

**Comment by gizmocuz** [comment]
That would be great! I send you a PM!

**Comment by ghost** [comment]
Send the qubino dimmer tot gizmocuz today

**Comment by YcKe** [comment]
Thanks a lot for the cooperation gentleman!

**Comment by bueno79** [comment]
Thank you so much. I appreciate a lot this community and his implications in this awesome project.

**Comment by gizmocuz** [comment]
Seems there is/was no problem with the switch, all is working perfectly.
Plz double check your groups/network, and make sure only node 1 (the controller) is in the lifeline group of all nodes.
Also make sure no scripts are doing strange things to the lights, disable the scripts for example

**Comment by AntoonvdOetelaar** [comment]
I have the same problem. I have a test setup for my new house. Yesterday I updated domoticz to the new stable (win7). No change. The real dimmer works fine. It's only when I push the switch in the domoticz GUI, in my case the dim level goes back to 9% (Why because a fibaro keeps the level the same) the real lamp does dim to 0% (switched off) but the GUI switch(lamp) stays on. When I press the GUI switch a second time it does switch off in GUI, and the real lamp stays off. My test setup only has a fibaro dimmer, a qubino dimmer, a qubino relay and a aeon labs multisensor 6.
No scripts no groups, my zwave network controller talks with every node directly and all to each other. (Perfect condition)

Could the minimum dim level in the node have an impact, what if I set it to 15%?
